cmd/gateway-gcs.go

```diff
@@ -459,11 +458,14 @@ func (l *gcsGateway) GetObjectInfo(bucket string, object string) (ObjectInfo, er
 	}
 
 	attrs, err := l.client.Bucket(bucket).Object(object).Attrs(l.ctx)
+
 	if err != nil {
 		return ObjectInfo{}, gcsToObjectError(traceError(err), bucket, object)
 	}
+	objInfo := fromGCSObjectInfo(attrs)
+	objInfo.ETag = fmt.Sprintf("%d", attrs.CRC32C)
```

Comment:
As @krishnasrinivas suggested we could potentially move this to `fromGCSObjectInfo()` function itself. where we will pick like this. 

Also it looks like the function name is wrong it should be `toGCSObjectInfo` not `fromGCSObjectInfo`

This can be combined it should work for all scenarios.
```
// toGCSObjectInfo converts GCS BucketAttrs to gateway ObjectInfo                                                                              
 func toGCSObjectInfo(attrs *storage.ObjectAttrs) ObjectInfo {
         objInfo := ObjectInfo{
                 Name:            attrs.Name,
                 Bucket:          attrs.Bucket,
                 ModTime:         attrs.Updated,
-                Size:            attrs.Size,
                 UserDefined:     attrs.Metadata,
                 ContentType:     attrs.ContentType,
                 ContentEncoding: attrs.ContentEncoding,
         }
+        objInfo.ETag = fmt.Sprintf("%d", attrs.CRC32C)
+        if attrs.MD5 != nil {
+                objInfo.ETag = hex.EncodeToString(attrs.MD5)
+        }
+        return	objInfo
 }
}
```